src/tracks/gosling-track/gosling-track.ts

```diff
@@ -333,14 +333,43 @@ const factory: PluginTrackFactory<Tile, GoslingTrackOptions> = (HGC, context, op
         override drawTile(tile: Tile) {
             if (PRINT_RENDERING_CYCLE) console.warn('drawTile(tile)');
 
-            tile.drawnAtScale = this._xScale.copy(); // being used in `super.draw()`
-
+            /**
+             * If we don't have info about the tile, we can't draw anything.
+             */
             const tileInfo = this.#processedTileInfo[tile.tileId];
             if (!tileInfo) {
                 // We do not have a track model prepared to visualize
                 return;
             }
 
+            /**
+             * Add a copy of the track scale to the tile. The tile needs its own scale because we will use it to
+             * determine how much the tile has been stretched (if we are stretching the graphics)
+             */
+            if (!tile.drawnAtScale) {
+                // This is the first time this tile is being drawn
+                tile.drawnAtScale = this._xScale.copy();
+            }
+
+            /**
+             * For certain types of marks and layouts (linear), we can stretch the graphics to avoid redrawing
+             * This is much more performant than redrawing everything at every frame
+             */
+            const [graphicsXScale, graphicsXPos] = this.getXScaleAndOffset(tile.drawnAtScale);
+            if (this.#hasStretchableGraphics() && !this.#isTooStretched(graphicsXScale) && graphicsXScale !== 1) {
```

Comment:
On first render, the graphicsXScale is equal to 1. We don't to stretch because nothing has been drawn yet. I just added a variable to better reflect this. Hopefully it is more clear now!